### ISSUE TITLE
How can we implement this method on LSTM

### ISSUE THREAD

**Issue by KingS770234358** [kohpangwei/influence-release#14]

**Comment by Kunlun-Zhu** [comment]
I guess influence function will work extremely slow on language model since word vector is so hard to do the gradient, and you need to take care that some of the assumption made in the paper based on supervised learning. You need to make some adaption.
I believe there are some papers about the interpretability of the language model like LSTM and analysis of its data importance, might be more helpful.

**Comment by KingS770234358** [comment]
@Kunlun-Zhu 
I really appreciate it you made time to answer my question.
I'm fresh man learning interpretability of deep learning.
As you advised,I read some other papers about the interpretability of the language model.But most of them are from the perspective of  features of samples but not from the perspective of  the importance of training-sample.
Now I am eager to use this method on LSTM.
Might I ask you for some advice.
As you mention in the reply,how can we do the gradient for word vector because there is a embedding layer in my neural network.
I will appreciate it so much if you could give me some advice.

**Comment by Kunlun-Zhu** [comment]
Well sadly, I am currently trying to do similar research in language model, and haven't get a feasible idea yet.

**Comment by kohpangwei** [comment]
In case it's still relevant, here is a link to a paper that computes influence functions on language models:
https://arxiv.org/pdf/2005.06676.pdf

Other ways of efficiently computing/approximating influence:
https://arxiv.org/abs/2008.03703
https://arxiv.org/pdf/2002.08484.pdf
https://arxiv.org/pdf/2012.04207.pdf
